docs/migration-guide.md

```diff
@@ -58,171 +82,62 @@ To add Storybook to a project that isn’t currently using Storybook:
 
-#### Automatically detect MDX2 errors with a CLI tool
+```sh
+# Convert storiesOf to CSF
+npx storybook@latest migrate storiesof-to-csf --glob="**/*.stories.tsx" --parser=tsx
+```
 
-If your project contains MDX files, run the following command before starting up Storybook:
+Then, you can optionally convert CSF 2 to CSF 3 ([why should you do this?](https://storybook.js.org/blog/storybook-csf3-is-here/)):
 
 ```sh
-npx @hipster/mdx2-issue-checker
+# Convert CSF 2 to CSF 3
+npx storybook@latest migrate csf-2-to-3 --glob="**/*.stories.tsx" --parser=tsx
 ```
 
-This will go through every MDX file in the current working directory, and show you which files have errors:
-
-![Terminal output of mdx2-issue-checker, showing errors found](./assets/mdx2-issue-checker-errors.png)
+### Migrating from `storiesOf` to dynamically created stories
 
-#### Fix MDX2 errors iteratively
+If you are using `storiesOf` for its ability to dynamically create stories, you can build your own "storiesOf" implementation by leveraging the new [(experimental) indexer API](https://storybook.js.org/docs/api/main-config-indexers). A proof of concept of such an implementation can be seen in [this StackBlitz demo](https://stackblitz.com/edit/github-h2rgfk?file=README.md). See the demo's README.md for a deeper explanation of the implementation.
 
-The CLI only shows the first error per file, so you might need to run the checker iteratively. One way to streamline this process if you use VS Code is to:
+### Migrating from `*.stories.mdx` to CSF
 
-1. Install the [MDX VS Code extension](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx)
-2. Enable experimental IntelliSense support for MDX files in your user settings: `"mdx.experimentalLanguageServer": true`
+Storybook now requires that MDX pages now reference stories written in CSF, rather than the previous .stories.mdx hybrid approach. You can automatically convert your files using the following codemod (make sure to update the glob to fit your files):
 
-This shows the errors visually in your editor, which speeds things up a lot. Here's what it looks like to fix multiple errors in a file using the extension:
+```sh
+npx storybook@latest migrate mdx-to-csf --glob "src/**/*.stories.mdx"
+```
 
-![MDX errors showing in VS Code](./assets/mdx-vs-code-extension-errors.gif)
+You’ll also need to update your stories glob in .storybook/main.js to include the newly created .mdx and .stories.js files if it doesn’t already.
 
-For more information on this change, see the [migration notes](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#storystorev7-enabled-by-default).
+**Note:** this migration supports the Storybook 6 [“CSF stories with MDX docs”](https://github.com/storybookjs/storybook/blob/6e19f0fe426d58f0f7981a42c3d0b0384fab49b1/code/addons/docs/docs/recipes.md#csf-stories-with-mdx-docs) recipe.
```

Comment:
@kylegach we could potentially have this in a `Callout` component and push it upwards after the code snippet

Reply:
The Callout draws a bit _too_ much attention to such a minor detail, imo.